crates/rustc_codegen_spirv/src/codegen_cx/entry.rs

```diff
@@ -290,6 +296,9 @@ impl<'tcx> CodegenCx<'tcx> {
         // which we represent with `SpirvType::InterfaceBlock` (see its doc comment).
         // This "interface block" construct is also required for "runtime arrays".
         let is_unsized = self.lookup_type(value_spirv_type).sizeof(self).is_none();
+        let is_pair = matches!(entry_arg_abi.mode, PassMode::Pair(..));
+        // is_pair implies is_unsized (but is_unsized doesn't imply is_pair, for RuntimeArray<T>)
+        assert!(!is_pair || is_unsized);
```

Comment:
Can't this become really confusing if someone uses some type that just happens to be represented as a pair? I think you could replace `if is_pair {` below with `if is_unsized_with_len {` given:
```rust
let is_unsized_with_len = is_unsized && matches!(entry_arg_abi.mode, PassMode::Pair(..));
```